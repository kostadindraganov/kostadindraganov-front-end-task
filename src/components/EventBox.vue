<template>
  <div 
  class="event-box absolute z-10 bg-white border-l-4 border-pink-500"
  :style="style"
  >
    <h3 class="title">
     Event title
    </h3>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { Events } from '@/types/Events';

@Component
export default class EventBox extends Vue {
  @Prop({ 
    type: Object, 
    default: {} 
  }) private event!: Events;

get style() {
  return `top:${this.event.top}px; left:${this.event.left}px; width:${this.event.width}px; height:${this.event.height}px;`
}

}
</script>
