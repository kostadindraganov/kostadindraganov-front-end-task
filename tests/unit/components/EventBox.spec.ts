import { shallowMount } from '@vue/test-utils';
import EventBox from '@/components/EventBox.vue';

let wrapper;

const event = {
    startTime: 100,
    endTime: 300,
    width: 333,
    top: 0,
    left: 0,
    height: 240,
}

describe('Testing component EventBox', () => {
	describe('EventBox', () => {
        wrapper = shallowMount(EventBox, {
            propsData: {
                event: event
            }
        })

        it('EventBox is a Vue instance', () => {
            expect(wrapper.vm).toBeTruthy();
        })
        
        it('EventBox has proper layout', () => {
            expect(wrapper.html()).toMatchSnapshot();
        })

        it('Test prop event', () => {
            expect(wrapper.vm.event).toBe(event);
        })

        it('Test added classes style', () => {
            expect(wrapper.vm.style).toBe("top:0px; left:0px; width:333px; height:240px;")
        })

        it('Test has proper title', () => {
            expect(wrapper.find('.title').exists()).toBe(true)
            expect(wrapper.find('.title').text()).toBe('Event title');
        })
	});
});