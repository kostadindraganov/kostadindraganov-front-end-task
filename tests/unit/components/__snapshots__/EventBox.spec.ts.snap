// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component EventBox EventBox EventBox has proper layout 1`] = `
<div class="event-box absolute z-10 bg-white border-l-4 border-pink-500" style="top: 0px; left: 0px; width: 333px; height: 240px;">
  <h3 class="title">
    Event title
  </h3>
</div>
`;
